alibi_detect/od/backends/keops/knn.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+import torch
+from pykeops.torch import LazyTensor
+
+
+def cdist(X, Y):
+    return ((X - Y)**2).sum(-1).sqrt()
+
+
+class KNNKeops:
+    def score(X, x_ref, k, kernel=None):
+        ensemble = isinstance(k, (np.ndarray, list, tuple))
+        X = torch.as_tensor(X)
+        X_keops = LazyTensor(X[:, None, :])
+        x_ref_keops = LazyTensor(x_ref[None, :, :])
+        K = -kernel(X_keops, x_ref_keops) if kernel else cdist(X_keops, x_ref_keops)
+        ks = np.array(k) if ensemble else np.array([k])
+        bot_k_inds = K.argKmin(np.max(ks), dim=1)
+        all_knn_dists = (X[:, None, :] - x_ref[bot_k_inds][:, ks-1, :]).norm(dim=2)
```

Comment:
This defeats the purpose of using `keops`. You can leverage `keops`'s argKmin functionality. See [here](https://github.com/getkeops/keops/blob/main/pykeops/pykeops/tutorials/knn/plot_knn_torch.py) for more detail. Clear benchmarking is required in a simple example script.